# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/*/util-linux/nice-mount-paths.patch
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./mount/mount.c.orig	Wed Mar 21 10:07:59 2001
+++ ./mount/mount.c	Wed Mar 21 10:12:15 2001
@@ -1061,6 +1061,23 @@
   /* Substitute values in opts, if required */
   opts = usersubst(opts);
 
+  /* make the node a nice absolute path name */
+  if (node[0] && node[0]!='/') {
+    char *thisdir=get_current_dir_name();
+    char *oldnode=node;
+    if (chdir(node) == 0) {
+      /* we will never free this variable ... */
+      node=get_current_dir_name();
+      chdir(thisdir);
+      if (verbose) printf("mount: change '%s' to '%s'\n",oldnode,node);
+    }
+    free(thisdir);
+  }
+  if (strlen(node)>=3 && node[strlen(node)-1]=='/') {
+    if (verbose) printf("mount: remove the leading '/' in '%s'\n",node);
+    node[strlen(node)-1]=0;
+  }
+
   /* Merge the fstab and command line options.  */
   if (opts == NULL)
        opts = cmdlineopts;
